podcasts/New Search/Model/SearchResultsModel.swift

```diff
@@ -99,6 +103,19 @@ class SearchResultsModel: ObservableObject {
         isShowingLocalResultsOnly = true
     }
 
+    private func buildPlayedEpisodesUUIDs(_ episodes: [EpisodeSearchResult]) -> Set<String> {
+        if episodes.isEmpty {
+            return []
+        }
+        let uuids = episodes.map { $0.uuid }
+        return DataManager.sharedManager.findPlayedEpisodesBy(uuids: uuids)
```

Comment:
What do you think about adding `DataManager.sharedManager` as a DI?

```swift
    private var dataManager: DataManager

    init(analyticsHelper: SearchAnalyticsHelper = SearchAnalyticsHelper(source: .unknown), dataManager: DataManager = DataManager.sharedManager) {
        self.analyticsHelper = analyticsHelper
        self.dataManager = dataManager
    }
```

Reply:
Yeah just kept as it was used in the other methods. I would really love to use the DI property wrapper created by PointFree or the ones used by Tumblr (which doesn't contains other dependencies), but it needs to be opensourced